imports/api/questions.js

```diff
@@ -195,84 +200,83 @@ if (Meteor.isServer) {
     } else this.ready()
   })
 
-  // question library for a user (student can see those they own or created)
-  Meteor.publish('questions.library', function () {
+  Meteor.publish('questions.library', function (courseId) {
+
     if (this.userId) {
       const user = Meteor.users.findOne({_id: this.userId})
-      if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else if (user.isInstructorAnyCourse()) {
-        const courses = _.pluck(Courses.find({instructors: this.userId}).fetch(), '_id')
-        return Questions.find({
-          '$or': [{owner: this.userId}, {courseId: { '$in': courses }, approved: true}],
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
+      if (!user.isInstructor(courseId) && !user.isStudent(courseId) && !user.hasRole(ROLES.admin)) return this.ready()
+      let query = { 
+        sessionId: {$exists: false},
+        courseId: courseId,
+      }
+      if (user.hasGreaterRole(ROLES.prof)) {
+        query = _.extend({ 
+          approved: true, 
+          '$or': [{owner: this.userId, private: true}, {'$or': [{ private: false }, { private: {$exists: false}}]}], 
```

Comment:
I don't think we need the privacy check actually since both public, default, and private questions can reside in the user's library. We just need to make sure private is false or does not exist for the other libraries.